### PR TITLE
Changing KEEP_REPORTDIR to boolean

### LINKED ISSUE #3384: KEEP_REPORTDIR should be boolean
KEEP_REPORTDIR is boolean https://github.com/adoptium/aqa-tests/blob/098041520884daafaa3d0b7c30eab6ab93800f41/buildenv/jenkins/JenkinsfileBase#L66

However, we set it as String in testJobTemplate. 
https://github.com/adoptium/aqa-tests/blob/8a55c344d1b9847d4d083aaae4e903a675b659b5/buildenv/jenkins/testJobTemplate#L40
https://github.com/adoptium/aqa-tests/blob/8a55c344d1b9847d4d083aaae4e903a675b659b5/buildenv/jenkins/testJobTemplate#L417

ToDo:
- Verify if we can use KEEP_REPORTDIR as a boolean in Grinder
   -> run a passed test (i.e., jdk_math) with KEEP_REPORTDIR as boolean (true and false) and check if junit result file shows up accordingly 
- if so, update testJobTemplate to use boolean

### PR DESCRIPTION
**Pull request by amir-waheed**
This patch changes the KEEP_REPORTDIR to boolean from string in testJobTemplate

Fixes #3384 

Signed-off-by: Abeer Waheed amir2@ualberta.ca

### CODE REVIEW

**Comment by amir-waheed** [comment]
Here is the link for the test I ran with KEEP_REPORTDIR as boolean in Grinder_New. The test results successfully shows.
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_New/2/testReport/

File: buildenv/jenkins/testJobTemplate [hunk 1]
```diff
@@ -414,7 +414,7 @@ ARCH_OS_LIST.each { ARCH_OS ->
 							    sectionHeaderStyle(sectionHeaderHelpTextStyleCss)
 							}
 							booleanParam('KEEP_WORKSPACE', false, "Keep workspace on the machine")
-							stringParam('KEEP_REPORTDIR', KEEP_REPORTDIR, "Keep the test report dir if the test passes?")
+							booleanParam('KEEP_REPORTDIR', KEEP_REPORTDIR, "Keep the test report dir if the test passes?")
```
[the remark this answers is not in the document]
Comment (amir-waheed) [reply buildenv/jenkins/testJobTemplate]: Done.

File: buildenv/jenkins/testJobTemplate [hunk 2]
```diff
@@ -414,7 +414,7 @@ ARCH_OS_LIST.each { ARCH_OS ->
 							    sectionHeaderStyle(sectionHeaderHelpTextStyleCss)
 							}
 							booleanParam('KEEP_WORKSPACE', false, "Keep workspace on the machine")
-							stringParam('KEEP_REPORTDIR', KEEP_REPORTDIR, "Keep the test report dir if the test passes?")
+							booleanParam('KEEP_REPORTDIR', KEEP_REPORTDIR.toBoolean(), "TKG will keep the test report dir if the test passes? Set true for OpenJDK junit results")
```
Comment (smlambert) [inline buildenv/jenkins/testJobTemplate:417]:
```suggestion
							booleanParam('KEEP_REPORTDIR', KEEP_REPORTDIR.toBoolean(), "TKG will keep the test report dir if the test passes. Set true for OpenJDK junit results")
```

**Review by smlambert — APPROVED** [review]
LGTM with a minor request to remove the '?' since the sentence is no longer a question.

**Review by renfeiw — APPROVED** [review]
Thanks, LGTM

**Pull request merged**